atlas_densities/densities/fitting.py

```diff
@@ -196,7 +197,11 @@ def compute_average_intensity(
         restricted_mask = volume_mask
     else:
         restricted_mask = np.zeros_like(volume_mask)
-        restricted_mask[slices] = True
+        # remove slices indices outside the volume_mask
+        slices_ = np.copy(slices)
```

Comment:
can you add a test case where this gets changed?

Also, is using an numpy array needed?
Aren't you just doing:
```
slices = [s for s in slices if 0 <= s < volume_mask.shape[0]]
```
?